controllers/periodic_syncer.go

```diff
@@ -52,7 +66,42 @@ func (r *OrganizationReconciler) Sync(ctx context.Context) error {
 	return nil
 }
 
-func (r *OrganizationReconciler) syncGroup(ctx context.Context, g keycloak.Group, org *orgv1.Organization) (*orgv1.Organization, error) {
+func (r *PeriodicSyncer) syncGroup(ctx context.Context, g keycloak.Group, org *orgv1.Organization) (runtime.Object, error) {
+	const depth = 0
+	switch len(g.PathMembers()) - depth {
+	case 1:
+		return r.syncOrganization(ctx, g, org)
+	case 2:
+		return r.syncTeam(ctx, g)
+	}
+
+	return nil, fmt.Errorf("invalid group hierarchy `%s`", g.Path())
```

Comment:
Not sure if we should return an error in that case. I don't think more keycloak sub-groups are invalid, we just don't sync them to the cluster

Reply:
I'd rather throw if we don't support something than do nothing...

Reply:
But yeah i see your point.

Reply:
I'll log an error but won't abort the sync. I feel like that's more useful here.